projects/web-wallet/rpc-protocol.md

```diff
@@ -0,0 +1,432 @@
+  1. `hash` - Hash of the transaction being spent in this input. Used if not type query
+  1. `index` - Index of the transction being spent in this input. Used if not type query
+  1. `max_utxos` - Maximum number of utxos to filter in the query. Optional query parameter when using type query
+  1. `token` - Token uid to filter utxos in the query. Optional query parameter when using type query 
+  1. `filter_address` - Address to filter utxos in the query. Optional query parameter when using type query 
+  1. `amount_smaller_than` - Filter only utxos with values smaller than this. Optional query parameter when using type query 
+  1. `amount_bigger_than` - Filter only utxos with value bigger than this. Optional query parameter when using type query.
+3. `changeAddress` - (Optional) Address to send the change
+
+
+**Example Requests:**
+
+```json
+{
+  "id": 3,
+  "jsonrpc": "2.0",
+  "method": "htr_sendTx",
+  "params": {
+  	"outputs": [{
+  	  "address": "H8RmX1AMQKAhWkBvf8DaKoAU7ph3Yiqg3c",
+  	  "value": 100,
+  	  "token": "00"
+  	}],
+    "inputs": [{
+      "hash": "00004788e55b9b3fb90aaad2065e99ccf36ccd2fac5ed40ae62eeefd2486e96c",
+      "index": 1
+    }]
+	}
+}
+```
+
+```json
+{
+  "id": 3,
+  "jsonrpc": "2.0",
+  "method": "htr_sendTx",
+  "params": {
+  	"outputs": [{
+  	  "address": "H8RmX1AMQKAhWkBvf8DaKoAU7ph3Yiqg3c",
+  	  "value": 100,
+  	  "token": "00"
+  	}],
+    "inputs": [{
+      "filter_address": "H8RmX1AMQKAhWkBvf8DaKoAU7ph3Yiqg3c",
+      "token": "00",
+      "max_utxos": 10,
+      "amount_bigger_than": 100
+    }]
+	}
+}
+```
+
+**Example Response:**
+
+```json
+{
+  "hash": "00000000059dfb65633acacc402c881b128cc7f5c04b6cea537ea2136f1b97fb",
+  "nonce": 2455281664,
+  "timestamp": 1594955941,
+  "version": 1,
+  "weight": 18.11897634891149,
+  "parents": [
+    "00000000556bbfee6d37cc099a17747b06f48ca3d9bf4af85c707aa95ad04b3f",
+    "00000000e2e3e304e364edebff1c04c95cc9ef282463295f6e417b85fec361dd"
+  ],
+  "inputs": [
+    {
+      "tx_id": "00000000caaa37ab729805b91af2de8174e3ef24410f4effc4ffda3b610eae65",
+      "index": 1,
+      "data": "RjBEAiAYR8jc+zqY596QyMp+K3Eag3kQB5aXdfYja19Fa17u0wIgCdhBQpjlBiAawP/9WRAqAzW85CJlBpzq+YVhUALg8IUhAueFQuEkAo+s2m7nj/hnh0nyphcUuxa2LoRBjOsEOHRQ"
+    },
+    {
+      "tx_id": "00000000caaa37ab729805b91af2de8174e3ef24410f4effc4ffda3b610eae65",
+      "index": 2,
+      "data": "RzBFAiEAofVXnCKNCEu4GRk7j+wHpQM6qmezRcfxHCe/PcUdbegCIE2nip27ZQtkpkEgNEhycqHM4CkLYMLVUgskphYsd/M9IQLHG6YJxXifQ6eMxPHbINFEJAUvrzKWe9V7AXXW4iywjg=="
+    }
+  ],
+  "outputs": [
+    {
+      "value": 100,
+      "token_data": 0,
+      "script": "dqkUqdK8VisGSJuNItIBRYFfSHfHjPeIrA=="
+    },
+    {
+      "value": 200,
+      "token_data": 0,
+      "script": "dqkUISAnpOn9Vo269QBvOfBeWJTLx82IrA=="
+    }
+  ],
+  "tokens": []
+}
+```
+
+
+### 2.1.2 `htr_createToken`
+
+The `htr_createToken` method creates a new token transaction based on parameters 
```

Comment:
Should we allow the caller to choose the inputs? Maybe a `push_tx: bool` parameter to allow the caller better control?

Reply:
Can't the dApp request a ready-to-push transaction?